lib/ansible/utils/plugin_docs.py

```diff
@@ -271,3 +271,81 @@ def get_versioned_doclink(path):
         return '{0}{1}/{2}'.format(base_url, doc_version, path)
     except Exception as ex:
         return '(unable to create versioned doc link for path {0}: {1})'.format(path, to_native(ex))
+
+
+def _find_adjacent(path, plugin, extensions):
+
+    found = None
+    adjacent = Path(path)
+
+    plugin_base_name = plugin.split('.')[-1]
+    if adjacent.stem != plugin_base_name:
+        # this should only affect filters/tests
+        adjacent = adjacent.with_name(plugin_base_name)
+
+    for ext in extensions:
+        candidate = adjacent.with_suffix(ext)
+        if candidate.exists():
+            found = to_native(candidate)
+            break
+
+    return found
+
+
+def find_plugin_docfile(plugin, plugin_type, loader):
+    '''  if the plugin lives in a non-python file (eg, win_X.ps1), require the corresponding 'sidecar' file for docs '''
+
+    context = loader.find_plugin_with_context(plugin, ignore_deprecated=False, check_aliases=True)
+    if not context or not context.resolved:
```

Comment:
The comment says this should only be true for filter/tests, but can happen for any plugin type.

```suggestion
    if (not context or not context.resolved) and plugin_type in ('filter', 'test'):
```